service/r2/routes.go

```diff
@@ -271,3 +271,27 @@ func fetchRollupRuleHistory(s *service, r *http.Request) (data interface{}, err
 	}
 	return models.NewRollupRuleSnapshots(hist), nil
 }
+
+func updateRuleSet(s *service, r *http.Request) (data interface{}, err error) {
```

Comment:
nit: Can we please move this to below `validateRuleSet`? I tend to group similar methods together so they are easier to find.